src/schema/v2/createDiscoveryArtworkReferenceMutation.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+  GraphQLEnumType,
+} from "graphql"
+import { ResolverContext } from "types/graphql"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { generateUuid, generateBeacon } from "./discoverArtworks"
+
+export const ReferenceTypesEnum = new GraphQLEnumType({
+  name: "ReferenceTypes",
+  values: {
+    LIKED_ARTWORKS: {
+      value: "likedArtworks",
+    },
+    DISLIKED_ARTWORKS: {
+      value: "dislikedArtworks",
+    },
+  },
+})
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DiscoveryArtworkReferenceMutationSuccess",
+  isTypeOf: (data) => data.success,
+  fields: () => ({
+    success: {
+      type: GraphQLBoolean,
+      resolve: (result) => result.success,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DiscoveryArtworkReferenceMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "CreateDiscoveryArtworkReferenceResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const CreateDiscoveryLikedArtworkMutation = mutationWithClientMutationId<
+  { userId: string; artworkId: string; reference: string },
+  any,
+  ResolverContext
+>({
+  name: "CreateDiscoveryArtworkReferenceMutation",
+  description: "Creates a cross reference for artwork and user in weaviate",
+  inputFields: {
+    userId: {
```

Comment:
don't think we need this here, it will increase some verbosity in the clients where they will have to get the userId before querying/mutating. We can just user the id coming from the header.

Reply:
thanks for the early feedback and good point! We can use as non-required field while we are still in research phase.